src/app/trainee/revalidation-table/revalidation-table.component.html

```diff
@@ -0,0 +1,85 @@
+            </td>
+        </ng-container>
+        <ng-container matColumnDef="outcome">
+            <th mat-header-cell scope="col" *matHeaderCellDef>Outcome</th>
+            <td mat-cell *matCellDef="let element">
+                <div class="outcome" [ngSwitch]="element.outcome">
+                    <mat-icon *ngSwitchCase="'Approved'" aria-hidden="true" class="icon-green">done_all
+                    </mat-icon>
+                    <mat-icon *ngSwitchCase="'Rejected'" aria-hidden="true" class="icon-red">error_outline
+                    </mat-icon>
+                    <mat-icon *ngSwitchCase="'Under review'" aria-hidden="true" class="icon-blue">done
+                    </mat-icon>
+                    <span class="outcome-text">{{ element.outcome }}</span>
+                </div>
+            </td>
+        </ng-container>
+        <ng-container matColumnDef="gmcSubDueDate">
+            <th mat-header-cell scope="col" *matHeaderCellDef>
+                GMC Submission Due Date
+            </th>
+            <td mat-cell *matCellDef="let element">
+                {{ element.gmcSubDueDate | date: dateFormat }}
+            </td>
+        </ng-container>
+        <ng-container matColumnDef="actSubDate">
+            <th mat-header-cell scope="col" *matHeaderCellDef>
+                Actual Submission Date
+            </th>
+            <td mat-cell *matCellDef="let element">
+                {{ element.actSubDate | date: dateFormat }}
+            </td>
+        </ng-container>
+        <ng-container matColumnDef="submittedBy">
+            <th mat-header-cell scope="col" *matHeaderCellDef>Submitted By</th>
+            <td mat-cell *matCellDef="let element">{{ element.submittedBy }}</td>
+        </ng-container>
+
+        <ng-container matColumnDef="expandedDetail">
+            <td mat-cell *matCellDef="let element" [attr.colspan]="columnsToDisplay.length">
+                <div class="element-detail" [@detailExpand]="
+            element == expandedElement ? 'expanded' : 'collapsed'
+          ">
+                    <div class="element-submission-status">
+                        <strong>Submission Status:</strong>
+                        {{ element.submissionStatus }}
+                    </div>
+                    <mat-divider *ngIf="element.comments?.length > 0"></mat-divider>
+                    <ng-container *ngFor="let comment of element.comments">
+                        <div class="element-notes">
+                            {{ comment.comment }}
+                        </div>
+                    </ng-container>
+                </div>
+            </td>
+        </ng-container>
+
+        <tr mat-header-row *matHeaderRowDef="columnsToDisplay"></tr>
+        <tr tabindex="0" mat-row *matRowDef="let element; columns: columnsToDisplay" class="element-row"
+            [class.expanded-row]="expandedElement === element" (click)="
+        expandedElement = expandedElement === element ? null : element
```

Comment:
could this be simplified a little or perhaps thrown in a method as its used several times in this template?

Reply:
done

Reply:
done